BLSQ/iaso
# IA-3685 IA-811: Allow filtering groups by project and by source and version

Related JIRA tickets : [IA-3685](https://bluesquare.atlassian.net/browse/IA-3685), [IA-811](https://bluesquare.atlassian.net/browse/IA-811)

## Self proofreading checklist


## Doc

N/A

## Changes

Added filters for project, data source and source version on groups page.

## How to test

From setuper folder, launch `python3 setuper.py --additional_projects` to create data and login with the created account.
Then check the groups page: http://localhost:8081/dashboard/orgunits/configuration/groups/.
Try to apply filters(Projects, Data Sources and Source version).

## Print screen / video

[Filter_by_Project_source_version.webm](https://github.com/user-attachments/assets/144d8ed7-7b31-46d1-99cf-8205698566d5)


## Notes

Things that the reviewers should know:

- known bugs that are out of the scope of the PR
- other trade-offs that were made
- does the PR depends on a PR in [bluesquare-components](https://github.com/BLSQ/bluesquare-components)?
- should the PR be merged into another PR?

## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: empty instance pop up

Refs: IA-3665
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.


[IA-3685]: https://bluesquare.atlassian.net/browse/IA-3685?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[IA-811]: https://bluesquare.atlassian.net/browse/IA-811?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
